### PR TITLE
fix: old chat cache still exists even after unmount

### PR DESCRIPTION
**Pull request by fajarism**
# Helio PR Checklist

## Please review all of the following checks before making PR

- [x] Jika menambahkan sebuah komponen, apakah sudah dipastikan tidak ada komponen yang redundant, baik secara UI maupun fungsional?
- [x] Periksa apakah komponen yang dibuat reusable (contoh: membuat komponen View untuk membuat Divider antar section)
- [x] Periksa untuk penggunaan state management yang baik dan benar (contoh: Context, useState), terutama penggunaan local state untuk optimistic UI (Pengoperan state dan data melalui props dari parent ke children dan sebaliknya)
- [x] Pastikan komponen yang dibuat sudah sesuai dengan module yang ada terutama penamaan dan penempatan untuk meningkatkan readability dan maintainability.
- [x] Implementasi fallback error untuk menangani kesalahan tak terduga, mencegah aplikasi mengalami crash
- [x] Evaluasi kompleksitas kode dan perbaiki logika yang kompleks dengan memecahnya menjadi fungsi atau komponen yang lebih kecil dan lebih mudah dimengerti.
- [x] Apakah sudah resolve semua komentar dari CodeRabbit🐇?
- [x] Apakah semua unit tests sudah Pass/Success? (Tidak ada bypass)
- [x] Periksa semua perubahan teks Bahasa Inggris approved oleh Bastian?
- [x] Apakah sudah menggunakan normalized.dimen/ normalizeFontSize untuk semua perubahan style?
- [x] Pastikan untuk mengikuti [guidelines pembuatan komponen Guidelines About Loading and Layout](https://docs.google.com/document/d/1GHbvEtFrQmEQXYBV9dLrPI654n2DbPwS1qjedajLSiw/edit)

### Summary by CodeRabbit

---
- Bug Fix: Updated the `useQuery` hook in `chatScreenHooks.js` to include `selectedChannelId` for unique cache keys, preventing message mix-ups across different channels.
- Refactor: Enhanced the `useChatScreenHook` function in `useChatScreenHook.ts` by adding a new state variable `setSelectedChannel` and implementing cleanup effects to reset chats and selected channel, improving chat session isolation and user experience.
---

### CODE REVIEW

File: src/hooks/screen/services/chatScreenHooks.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import {useQuery} from 'react-query';
+
 import ChatSchema from '../../../database/schema/ChatSchema';
 
 export function useGetAllMessage(payload, options) {
   return useQuery(
-    ['getAllMesage', payload],
+    [`getAllMesage-${payload?.selectedChannelId}`, payload],
```
Comment (bastian-purrer) [inline src/hooks/screen/services/chatScreenHooks.js:7]: Image description   CodeRabbit

The change made to the key in `useQuery` is a good one. It ensures that each channel has its own unique cache, which can help prevent data from different channels from getting mixed up. However, there's a typo in the string 'getAllMesage'. It should be 'getAllMessage'.
```diff
-    [`getAllMesage-${payload?.selectedChannelId}`, payload],
+    [`getAllMessage-${payload?.selectedChannelId}`, payload],
```

File: src/hooks/screen/useChatScreenHook.ts [hunk 2]
```diff
@@ -35,8 +35,13 @@ export const ScrollContext = React.createContext<ScrollContextProps | null>(null
...
   const {localDb, refresh, otherListener} = useLocalDatabaseHook();
-  const {selectedChannel, goBackFromChatScreen, goToChatInfoScreen, setChannelAsRead} =
-    useChatUtilsHook(type);
+  const {
+    selectedChannel,
+    goBackFromChatScreen,
+    goToChatInfoScreen,
+    setChannelAsRead,
+    setSelectedChannel
+  } = useChatUtilsHook(type);
```
Comment (bastian-purrer) [inline src/hooks/screen/useChatScreenHook.ts:44]: Image description   CodeRabbit

The addition of `setSelectedChannel` from `useChatUtilsHook(type)` seems to be a good change. It's used later in the cleanup function of a `useEffect` hook, which is a common pattern for resetting state when a component unmounts.

```typescript
const {
  selectedChannel,
  goBackFromChatScreen,
  goToChatInfoScreen,
  setChannelAsRead,
  setSelectedChannel
} = useChatUtilsHook(type);
```

File: src/hooks/screen/useChatScreenHook.ts [hunk 3]
```diff
@@ -313,6 +318,7 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
   React.useEffect(() => {
     return () => {
       setChats([]);
+      setSelectedChannel(null);
```
Comment (bastian-purrer) [inline src/hooks/screen/useChatScreenHook.ts:321]: Image description   CodeRabbit

This line is part of the cleanup function in a `useEffect` hook. The call to `setSelectedChannel(null)` is a good practice as it resets the state when the component unmounts, preventing potential memory leaks or stale data issues.

```typescript
setSelectedChannel(null);
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between b82669832d4a0078be7ae9773589a51fa5c19001 and 678f571a2b09c6eb32f70e8cfa7c909f3e38d3b4 commits.
</details>

<details>
<summary>Files selected (2)</summary>

* src/hooks/screen/services/chatScreenHooks.js (1)
* src/hooks/screen/useChatScreenHook.ts (2)
</details>








<details>
<summary>Review comments generated (3)</summary>

* Review: 3
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by alamsyaah — APPROVED** [review]

**Pull request merged**